packages/go/analysis/ad/queries.go

```diff
@@ -653,9 +653,17 @@ func FetchDCSyncerPaths(tx graph.Transaction, node *graph.Node) (graph.PathSet,
 }
 
 func FetchForeignGPOControllers(tx graph.Transaction, node *graph.Node, skip, limit int) (graph.NodeSet, error) {
-	if domainSID, err := node.Properties.Get(ad.DomainSID.String()).String(); err != nil {
-		return nil, err
-	} else if gpoIDs, err := ops.FetchNodeIDs(tx.Nodes().Filterf(func() graph.Criteria {
+	var domainSID string
+	if sid, err := node.Properties.Get(ad.DomainSID.String()).String(); err != nil {
+		if sid, err := node.Properties.Get(common.ObjectID.String()).String(); err != nil {
+			return nil, err
+		} else {
+			domainSID = sid
+		}
+	} else {
+		domainSID = sid
+	}
```

Comment:
Falling back to the object id is clever! 

It looks like we are probably even getting the domain sid upon ingest but then throw it away because of the narrow options for providing details about incoming relationships. Ideally we would fix the root cause but that may be out of scope for this work. 

Since this pattern is used multiple times in all of these foreign 'x' functions I would suggest creating a small function to abstract this logic so there is less code repetition. Then it would also be easily unit testable :D

Let me know if you have any questions. Happy to clarify anything.

Reply:
I had been considering splitting it out, but I decided against it since it was only for a few functions and was pretty "generic" code. However, since you think it's a good idea as well I went ahead and split it out! I kinda wish we could just do something like `domainSID := Node.Properties.DomainSID || Node.Properties.ObjectID` 😆